lib/OpenQA/CacheService/Model/Cache.pm

```diff
@@ -47,139 +48,106 @@ sub from_worker {
-                        if ($progress < $current) {
-                            $progress = $current;
-                            log_debug "CACHE: Downloading $asset: " . ($size == $len ? 100 : $progress) . "%";
-                        }
-                    }
-                });
-        });
+    # Assets might be deleted by a sysadmin
+    my $tx = $ua->build_tx(GET => $url);
+    $tx->req->headers->header('If-None-Match' => qq{$etag}) if $etag && -e $asset;
 
     $tx = $ua->start($tx);
-    my $code = ($tx->res->code) ? $tx->res->code : 521;    # Used by cloudflare to indicate web server is down.
-    my $size;
+    my $res = $tx->res;
+    my $ret;
+
+    my $code = $res->code // 521;    # Used by cloudflare to indicate web server is down.
     if ($code eq 304) {
-        if ($self->_update_asset_last_use($asset)) {
-            log_debug "CACHE: Content has not changed, not downloading the $asset but updating last use";
-        }
-        else {
-            log_debug 'CACHE: Abnormal situation, code 304. Retrying download';
-            $asset = 520;
+        $log->info("Content of $asset has not changed, updating last use");
+        unless ($self->_update_asset_last_use($asset)) {
+            $log->warn('Abnormal situation, status 304');
+            $ret = 520;
         }
     }
-    elsif ($tx->res->is_server_error) {
-        log_debug "CACHE: Could not download the asset, triggering a retry for $code.";
-        log_debug "CACHE: Abnormal situation, server error. Retrying download";
-        $asset = $code;
+
+    elsif ($res->is_server_error) {
+        $log->warn("Abnormal situation, server error $code");
+        $ret = $code;
     }
-    elsif ($tx->res->is_success) {
+
+    elsif ($res->is_success) {
+        my $headers = $tx->res->headers;
         $etag = $headers->etag;
-        unlink($asset);
+        unlink $asset;
         $self->_cache_sync;
-        my $size = $tx->res->content->asset->move_to($asset)->size;
+        my $size = $res->content->asset->move_to($asset)->size;
         if ($size == $headers->content_length) {
             $self->_check_limits($size);
 
             # This needs to go in to the database at any cost - we have the lock and we succeeded in download
             # We can't just throw it away if database locks.
             my $att = 0;
             my $ok;
-            ++$att and sleep 1 and log_debug("CACHE: Error updating Cache: attempting again: $att")
+            ++$att and sleep 1 and $log->warn("Updating cache failed (attempt $att)")
               until ($ok = $self->_update_asset($asset, $etag, $size)) || $att > 5;
-            unless ($ok) {
-                log_error 'CACHE: FAIL Could not update DB - purging asset';
+
+            if ($ok) { $log->info("Download of $asset successful, new cache size is $self->{cache_real_size}") }
+            else {
+                $log->error("Purging $asset because updating the cache failed, this should never happen");
```

Comment:
Not sure whether the ", this should never happen" part is worth much when spotting this within the log after all.

Reply:
I think it is, that would mean the SQLite file was locked for 5 tries, which would mean we have very serious locking issues in the database. I really never want to see that error in the logs. :)